### PR TITLE
PAYARA-3087 Add support for property replacement for ${MPCONFIG=value…

### LINKED ISSUE #3085: Docker secrets and @DataSourceDefinitions in Payara Micro
# Description #
----------

I'm using the Payara Micro Docker image and injecting a Docker secret into the file `/run/secrets/database-password`. I would like to use that secret in a `@DataSourceDefinition` annotation.
```Java
@DataSourceDefinition(name = "java:app/MyDataSource",
        minPoolSize = 4,
        maxPoolSize = 8,
        className = "org.postgresql.ds.PGPoolingDataSource",
        url = "${dataSource.url}",
        user = "${dataSource.user}",
        password = "${ALIAS=dataSource.password}"
```
If I was using Payara Server I would run `asadmin --passwordfile /run/secrets/database-password create-password-alias dataSource.password` but I don't believe any of the --postbootcommandfile, --postdeploycommandfile and --prebootcommandfile options support the --passwordfile option?

I noticed there was a --secretsdir option but it seems restricted to Microprofile Config and I'm unsure how to get it from there into my `@DataSourceDefinition`.

## Expected Outcome

Perhaps make `${ALIAS=dataSource.password}` look in the --secretsdir or if it's already possible provide an example in the Payara Micro Docker image description at https://hub.docker.com/r/payara/micro?

## Current Outcome

Without a mechanism in place it becomes difficult to use Dockers recommended practices for managing secrets when connecting to datasources. All of the examples I've seen so far are either using hardcoded credentials or passing environment variables.

## Environment ##

- **Payara Version**: 5.182
- **Edition**: Micro
- **JDK Version**: 8 u151 - OpenJDK
- **Operating System**: Linux
- **Database**:

### PR DESCRIPTION
**Pull request by smillidge**
…} from server level MP Config Sources e.g. SecretsDir

Does not support replacement using application deployed ConfigSources or properties.
Fixes #3085

### CODE REVIEW

**Comment by smillidge** [comment]
jenkins test please

File: appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java [hunk 1]
```diff
@@ -1360,6 +1360,10 @@ private JdbcConnectionPool getJdbcConnectionPool(javax.resource.spi.ManagedConne
     }
 
     private boolean isSlowQueryLoggingEnabled() {
-        return !IS_SLOW_SQL_LOGGING_DISABLED.equals(connectionPool.getSlowQueryThresholdInSeconds());
+        if (connectionPool == null) {
+            return false;
+        } else {
```
Comment (Pandrex247) [inline appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/spi/ManagedConnectionImpl.java]: This change doesn't seem related to the rest of the PR?

**Comment by smillidge** [comment]
jenkins test please

File: nucleus/admin/config-api/pom.xml [hunk 2]
```diff
@@ -180,6 +180,12 @@
             <artifactId>javax.el</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
+            <groupId>org.eclipse.microprofile.config</groupId>
+            <artifactId>microprofile-config-api</artifactId>
+            <version>1.3</version>
```
Comment (Pandrex247) [inline nucleus/admin/config-api/pom.xml]: Can you use a property please.

**Comment by smillidge** [comment]
jenkins test please
